numba/tests/test_builtins.py

```diff
@@ -542,7 +542,7 @@ def test_float(self, flags=forceobj_flags):
             self.assertPreciseEqual(cfunc(x), pyfunc(x), prec='single')
 
         cfunc = jit((types.string,), **flags)(pyfunc)
```

Comment:
Is this test using object mode to pass? A string literal seems to be required for your `float("inf")` implementation, so I'm having trouble seeing how else this could be working.

Reply:
Hi Graham, I added a new test case with object mode disabled. The current implementation is specific for `float(+/- 'inf')` and if you think it is needed, I can work on arbitrary string -> float conversion.